backend/Bindings/Edna.Bindings.LtiAdvantage/Services/NrpsClient.cs

```diff
@@ -70,18 +71,27 @@ public async Task<IEnumerable<Member>> Get(string clientId, string tokenUrl, str
             httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(Constants.MediaTypes.MembershipContainer));
 
             _logger.LogInformation("Getting members.");
-            using var response = await httpClient.GetAsync(membershipUrl);
+            List<Member> members = new List<Member>();
 
-            if (!response.IsSuccessStatusCode)
+            do
             {
-                _logger.LogError("Could not get members.");
-                throw new Exception(response.ReasonPhrase);
+                using var response = await httpClient.GetAsync(membershipUrl);

```

Comment:
This seems to be making the calls one after the other which could lead to slowing down the overall response time of the API, how much latency increases this causes in the API? Can we check if there's a way to parameterize the NRPS request, so that we can increase the limit of students? The limit can then be changed as azure function parameters to allow different schools to set it according to their needs 🤔

Reply:
The spec says that the tool may send a parameter to specify the list size, but the platform may send more/less than the specified value, and that it is on the tool to handle this case with the next-link-header. So handling this case. But yes we can think of the above suggested way in case it becomes a bottleneck for users. The latency increase now will depend on the chunk size that the platform sends and the course size.

Reply:
Sure. I was primarily worried about the latency of the API since it'll be calling downstream APIs one after the other. If the latency increase is not too much, I'm ok with the current design.